### DISCUSSION TITLE
How to use StorageClass in one k8s to link to Ceph cluster outside the current k8s

### DISCUSSION THREAD

**Discussion by wangchao6856** [rook/rook#14526] [Q&A] [answered]
There is a Ceph cluster deployed on physical machines (installed via RPM). Now, we need to use StorageClass on Kubernetes to access this Ceph cluster.

1. Do I need to install Rook on the current Kubernetes and configure it in external cluster mode? If so, I have read the official documentation(`https://rook.github.io/docs/rook/latest-release/CRDs/Cluster/external-cluster/external-cluster/#prerequisites`) but there are some parts I don't understand:
How do I modify the chart source yaml to directly enable external mode when deploying Rook (without its own storage, directly pointing to external storage)? From the information I have seen, the cluster part uses `values-external.yaml`, but where is the external Ceph information specified?
2. If it is possible to specify through StorageClass directly, how should StorageClass and Static PV be written?

**Comment by zhucan** [comment]
no need to install rook. only need to deploy ceph csi under k8s cluster https://github.com/ceph/ceph-csi

**Reply by wangchao6856** [reply]
Does the ceph-csi chart require additional configuration information? An error occurred when deploying the chart source code:
`Error: template: ceph-csi-cephfs/templates/nodeplugin-daemonset.yaml:89:73: executing "ceph-csi-cephfs/templates/nodeplugin-daemonset.yaml" at <.Values.readAffinity.enabled>: nil pointer evaluating interface {}.enabled `
I understand the error indicates that the `readAffinity.enabled` information needs to be specified in the values.yaml?

**Reply by Madhu-1** [reply]
@wangchao6856 you might be using older version of helm please check https://github.com/ceph/ceph-csi/pull/4716

**Reply by parth-gr** [reply]
I am just wondering who will create the csi pods in this case

**Reply by wangchao6856** [reply]
I downloaded the source code for versions v3.10.2 、v3.11.0 and devel version, and packaged them myself in the charts directory using Helm. Both versions encountered the same error

**Reply by wangchao6856** [reply]
I would like to know if there are any examples of deploying ceph-csi and configuring it with external Ceph cluster information.

**Reply by Madhu-1** [reply]
If you are not planning to use Rook and use standalone cephcsi, please open an in ceph-csi repo

**Reply by wangchao6856** [reply]
Is there a logical difference between downloading the chart source code and packaging it myself with helm package versus using the helm repo method? Also, my server cannot directly connect to the GitHub address.

**Reply by zhucan** [reply]
I think there is no any differences.

**Reply by wangchao6856** [reply, marked as answer]
Thank you all for your suggestions. I have successfully deployed it by using the v3.10.2 version of the chart source code from csph-csi to create my own Helm package, and I have successfully mounted Ceph PV across Kubernetes clusters using StorageClass. However, please note that you need to remove `seLinuxMount: true` in `templates/csidriver-crd.yaml`, otherwise, an error will occur during installation.
